### PR TITLE
docs: use .env (and .env.example) for tokens used in release

### PR DESCRIPTION
**Pull request by zeke**
This PR adds a `.env.example` file to the repo which serves as a document of all the env vars required by the release scripts. The release scripts now run `require('dotenv-safe').load()` which loads the contents of `.env` into `process.env` and ensures that any human running commands like `npm run prepare-release` or `npm run release` will see helpful output if any of the required env is missing.

Eventually we should have the S3_* env vars in this file too, but for now I think those are still only used in python files. Is that correct, @jkleinsc

### CODE REVIEW

**Comment by zeke** [comment]
If it feels right, we could use https://github.com/theskumar/python-dotenv load S3 vars in the interim before we rewrite the remaining python scripts in JS.

File: .env.example [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# See docs/development/releasing.md
+
+APPVEYOR_TOKEN=
+CIRCLE_TOKEN=
+ELECTRON_GITHUB_TOKEN=
+VSTS_TOKEN=
```
Comment (ckerr) [inline .env.example]: Missing newline at EOF

Comment (ckerr) [inline .env.example]: The other non-S3 environment variable in my notes is `CHANGELOG_GITHUB_TOKEN` but it wasn't clear to me if it's required for releasing -- another question for @jkleinsc

**Comment by ckerr** [comment]
Hmm. I still like the idea of codifying the environment variables in a single canonical place as dotenv-safe does, but John's got a good point about this needing to be something that works in python as well as ES...

File: .env.example (hunk 1 above)
Comment (jkleinsc) [reply .env.example]: CHANGELOG_GITHUB_TOKEN isn't needed.  That was something I was using to test an alternate changelog generator.

**Comment by zeke** [comment]
> The scripts already have checks for the environment variables, so its simply replacing the existing validation with another.

True, but the purpose of `.env.example` is two-fold: it serves as both documentation and validation.

Also, we could remove all the existing code checks in this PR.. 🤔 

> we could add a comment stating that the environment variables are only needed for publishing releases.

Done. 👍

**Comment by jkleinsc** [comment]
@zeke I'm good with this change if we remove the existing asserts that are checking those environment variables.

**Pull request merged**